Smoke me/jkeenan/gh 21393 unused var

jkeenan opened the pull request:
This suppresses the `-Wunused_variable` warning reported in https://github.com/Perl/perl5/issues/21393 -- but since I'm just monkey-patching, please review.

(You may get a porting test failure because (I think) Module-CoreList needs to be updated after today's release.)

jkeenan committed:
Suppress -Wunused_variable warning during 'make'

Implement Dave Mitchell's suggestion: we don't really need the unused
variable.

For: https://github.com/Perl/perl5/issues/21393

mauke commented on pp_ctl.c:
```diff
@@ -4452,6 +4452,7 @@ S_require_file(pTHX_ SV *sv)
         PUSHMARK(PL_stack_sp);
         rpp_xpush_1(name_sv); /* always use the object for method calls */
         int count = call_sv(PL_hook__require__before, G_SCALAR);
```
While we're at it, the type of `count` should be changed from `int` to `I32` because that's what `call_sv` returns.

jkeenan commented:
> Really the count variable and assertion should just be removed, and call_sv() be called without using it's return value. The code I changed had 'if (count && ...)', but call_sv(...,G_SCALAR) always returns 1. The assertion was just a temporary crutch to check that I wasn't changing behaviour when I removed the 'count > 0' check.

See latest version of this pull request.

iabyn commented:
LGTM
